tests/testing_utils.py

```diff
@@ -69,6 +85,23 @@ def is_bitsandbytes_multi_backend_available() -> bool:
 )
 
 
+def is_flash_attn_available():
+    flash_attn_available = is_flash_attn_2_available()
+    kernels_available = is_kernels_available()
+    try:
+        from kernels import get_kernel
+
+        get_kernel("kernels-community/flash-attn")
+    except Exception:
+        kernels_available = False
+
+    return kernels_available or flash_attn_available
+
+
+# Function ported from transformers.testing_utils
+require_flash_attn = pytest.mark.skipif(not is_flash_attn_available(), reason="test requires Flash Attention")
+
```

Comment:
I don't think this one it 100% correct, because if kernels is available, but not flash-attn, and you try to use `atto_implementation="flash_attention2"`, it will most likely fail.

Reply:
OK I see.
A lot of our tests rely on flash-attn lib; it's probably a good time to drop `flash-attn` and rely only on `kernels`:

- replace all `attn_implementation="flash_attention_2"` -> `attn_implementation="kernels-community/flash-attn"`
- replace `require_flash_attn` by `require_kernels`

@albertvillanova we can do this in a future PR

Reply:
I agree! I created an issue for that:
- #4319